hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1810,6 +1810,18 @@ def register_type_strategy(
             f"for {origin!r} which can inspect specific type objects and return a "
             "strategy."
         )
+    if "pydantic.generics" in sys.modules and (
+        issubclass(custom_type, sys.modules["pydantic.generics"].GenericModel)
+        and not re.search(r"[A-Za-z_]+\[.+\]", repr(custom_type))
```

Comment:
Just a quick question: do you think runtime performance of `re.search` is good enough here?

Reply:
Yes - it's a simple pattern on a small string, and `st.register_type_strategy()` is not called very often.